src/components/AddExpensesForm.js

```diff
@@ -0,0 +1,91 @@
+import { useState } from 'react';
+import { v4 as uuid } from 'uuid';
+
+const AddExpensesForm = () => {
+  const initialState = {
+    id: '',
+    name: '',
+    date: '',
+    amount: '',
+  };
+
+  const [singleExpense, setSingleExpense] = useState(initialState);
+  const [expenses, setExpenses] = useState([]);
+
+  const handleChange = (evt) => {
+    const { name, value } = evt.target;
+    setSingleExpense({
+      ...singleExpense,
+      [name]: value,
+    });
+  };
+
+  const handleSubmit = (evt) => {
+    evt.preventDefault();
+    const newExpenses = [...expenses, { ...singleExpense, id: uuid() }];
+    setExpenses(newExpenses);
+    setSingleExpense(initialState);
+  };
+
+  return (
+    <div className="container-md">
+      <div className="text-center">
+        <h2 className="display-2">Add some Expenses</h2>
+      </div>
+      <form className="row g-2" onSubmit={handleSubmit}>
+        <div className="col-md">
+          <div className="form-floating">
+            <input
+              className="form-control"
+              type="text"
+              id="name"
+              name="name"
+              required
+              onChange={handleChange}
+              value={singleExpense.name}
+              aria-label="Expense Name"
+              placeholder="Expense Name"
+            />
+            <label htmlFor="amount">Expense Name</label>
+          </div>
+        </div>
+        <div className="col-md">
+          <div className="form-floating">
+            <input
+              className="form-control"
+              type="date"
+              id="date"
+              name="date"
+              required
+              onChange={handleChange}
+              value={singleExpense.date}
+              aria-label="Expense date"
+            />
+            <label htmlFor="date">Expense Date</label>
+          </div>
+        </div>
+        <div className="form-floating mb-2">
+          <input
+            className="form-control"
+            type="number"
+            id="amount"
```

Comment:
why is the id needed?

Reply:
normally the id is needed to link the input to the label. The bootstrap docs show that we should use it also with `floating-labels` but does not provide any explanation. Honestly, I do not find any difference with or without. It might still be relevant or screen readers maybe...